Runtime 5.0.11 release notes (#1424)
astronomer/docs: software_versioned_docs/version-0.29/runtime-release-notes.md

For instructions on how to upgrade, read [Upgrade Airflow on Astronomer Software](manage-airflow-versions.md). For general product release notes, go to [Software release notes](release-notes.md). If you have any questions or a bug to report, reach out to [Astronomer support](https://support.astronomer.io).

## Astro Runtime 6.0.3

- Release date: October 24, 2022
- Airflow version: 2.4.2

### Airflow 2.4.2

> Astro Runtime 6.0.3 includes same-day support for Airflow 2.4.2. Some changes in Airflow 2.4.2 include:

Reviewer: ```suggestion
Astro Runtime 6.0.3 includes same-day support for Airflow 2.4.2. The following are some of the changes included in Airflow 2.4.2:
```

> - Handle mapped tasks in task duration chart ([#26722](https://github.com/apache/airflow/pull/26722))

Reviewer: ```suggestion
- Handle mapped tasks in task duration chart [#26722](https://github.com/apache/airflow/pull/26722).
```

> - Make tracebacks opt-in ([#27059](https://github.com/apache/airflow/pull/27059)

Reviewer: ```suggestion
- Make tracebacks opt-in [#27059](https://github.com/apache/airflow/pull/27059).
```

> For a complete list of commits, see the [Apache Airflow 2.4.2 release notes](https://airflow.apache.org/docs/apache-airflow/stable/release_notes.html#airflow-2-4-2-2022-10-23.

Reviewer: ```suggestion
For a complete list of commits, see the [Apache Airflow 2.4.2 release notes](https://airflow.apache.org/docs/apache-airflow/stable/release_notes.html#airflow-2-4-2-2022-10-23).
```

### Additional improvements  

- Upgraded `openlineage-airflow` to 0.15.1, which includes a dedicated Airflow development environment. You can now create and test changes to custom OpenLineage extractors in an Airflow environment without needing to rebuild your Docker images. For more information, see the [OpenLineage changelog](https://github.com/OpenLineage/OpenLineage/blob/main/CHANGELOG.md).

## Astro Runtime 6.0.2

- Release date: September 30, 2022
- Airflow version: 2.4.1

### Support for Apache Airflow 2.4.1

> Astro Runtime 6.0.2 includes Airflow 2.4.1, which includes only bug fixes. For a complete list of commits, see the [Apache Airflow 2.4.1 release notes](https://airflow.apache.org/docs/apache-airflow/stable/release_notes.html#airflow-2-4-1-2022-09-30)).

Reviewer: ```suggestion
Astro Runtime 6.0.2 includes Airflow 2.4.1, which is limited to bug fixes. For a complete list of commits, see the [Apache Airflow 2.4.1 release notes](https://airflow.apache.org/docs/apache-airflow/stable/release_notes.html#airflow-2-4-1-2022-09-30).
```
Trying to avoid the duplicate use of includes here. You can probably remove _which is limited to bug fixes_.

### Backported bug fixes from Apache Airflow 2.4.2

Astro Runtime 6.0.2 includes the following bug fixes:

> - Remove DAG parsing from StandardTaskRunner ([#26750](https://github.com/apache/airflow/pull/26750))

Reviewer: ```suggestion
- Remove DAG parsing from StandardTaskRunner [#26750](https://github.com/apache/airflow/pull/26750).
```

> - Fix airflow tasks run --local when dags_folder differs from that of processor ([#26509](https://github.com/apache/airflow/pull/26509))

Reviewer: ```suggestion
- Fix airflow tasks run `--local` when dags_folder differs from the processor [#26509](https://github.com/apache/airflow/pull/26509).
```

> - Add fixture for CLI tests requiring sample dags ([#26536](https://github.com/apache/airflow/pull/26536))

Reviewer: ```suggestion
- Add fixture for CLI tests requiring sample DAGs [#26536](https://github.com/apache/airflow/pull/26536).
```

- Backported a fix to correct an issue where logs were not loading from Celery workers ([#26493](https://github.com/apache/airflow/pull/26493))
- Fixed [CVE-2022-40674](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2022-40674)

> ## Astro Runtime 5.10.11

Reviewer: ```suggestion
## Astro Runtime 5.0.11
```
![image](https://user-images.githubusercontent.com/104205257/199714405-20c3698f-188f-4f0c-a099-9c00f4d4b16f.png)

- Release date: November 2, 2022
- Airflow version: 2.3.4 

### Backported Airflow bug fixes

> Astro Runtime 5.10.11 includes the following bug fix from later Apache Airflow releases:

Author: ```suggestion
Astro Runtime 5.0.11 includes the following bug fix from later Apache Airflow releases:
```

> Astro Runtime 5.10.11 includes the following bug fixes from later Apache Airflow releases:

Author: ```suggestion
Astro Runtime 5.10.11 includes the following bug fix from later Apache Airflow releases:
```

> - Fix warning when using xcomarg dependencies ([#26801](https://github.com/apache/airflow/pull/26801))

Reviewer: ```suggestion
- Fix warning when using XComArg dependencies [#26801](https://github.com/apache/airflow/pull/26801).
```

### Bug fixes

> - Removed the default value for `AIRFLOW__LOGGING__REMOTE_BASE_LOG_FOLDER`, as this value is now set the Astro data plane. This enables Astronomer Software users to set a value for custom remote logging storage solutions. 

Reviewer: ```suggestion
- Removed the default value for `AIRFLOW__LOGGING__REMOTE_BASE_LOG_FOLDER`, as this value is now set in the Astro data plane. This enables Astronomer Software users to set a value for custom remote logging storage solutions. 
```

> - Removed the default value for `AIRFLOW__LOGGING__REMOTE_BASE_LOG_FOLDER`, as this value is now set the Astro data plane. This enables Astronomer Software users to set the value for custom remote logging storage solutions. 

Author: ```suggestion
- Removed the default value for `AIRFLOW__LOGGING__REMOTE_BASE_LOG_FOLDER`, as this value is now set the Astro data plane. This enables Astronomer Software users to set a value for custom remote logging storage solutions. 
```

## Astro Runtime 5.0.10

- Release date: October 17, 2022
- Airflow version: 2.3.4

### Additional improvements

> - Upgraded `astronomer-providers` to 1.10.0, which includes two new deferrable versions of operators, `SFTPSensorAsync` and `ExternalDeploymentTaskSensorAsync`. See the [Astronomer Providers changelog](https://github.com/astronomer/astronomer-providers/blob/1.10.0/CHANGELOG.rst).

Reviewer: ```suggestion
- Upgraded `astronomer-providers` to 1.10.0, which includes two new deferrable versions of the operators `SFTPSensorAsync` and `ExternalDeploymentTaskSensorAsync`. See the [Astronomer Providers changelog](https://github.com/astronomer/astronomer-providers/blob/1.10.0/CHANGELOG.rst).
```

> - Upgraded `openlineage-airflow` to version `0.15.1`. See the [OpenLineage changelog](https://github.com/OpenLineage/OpenLineage/blob/main/CHANGELOG.md).

Reviewer: ```suggestion
- Upgraded `openlineage-airflow` to version 0.15.1. See the [OpenLineage changelog](https://github.com/OpenLineage/OpenLineage/blob/main/CHANGELOG.md).
```

### Bug fixes

> - Revert “Cache the custom secrets backend so the same instance gets re-used” ([#25556](https://github.com/apache/airflow/pull/25556))

Reviewer: ```suggestion
- Revert “Cache the custom secrets backend so the same instance gets reused” [#25556](https://github.com/apache/airflow/pull/25556).
```

> - Fixed faulty Kubernetes executor config serialization logic

Reviewer: ```suggestion
- Fixed faulty Kubernetes executor config serialization logic.
```

- Fixed an issue where logs were not loading from Celery workers ([#26337](https://github.com/apache/airflow/pull/26337) and [#26493](https://github.com/apache/airflow/pull/26493))
- Fixed CVE-2022-40754 ([#26409](https://github.com/apache/airflow/pull/26409))
- Fixed the Airflow UI not auto-refreshing when scheduled tasks are running. This bug was introduced in Airflow 2.3.4 ([#25950](https://github.com/apache/airflow/pull/25950))

> - Fixed an issue where the scheduler could crash when queueing dynamically mapped tasks ([#25788](https://github.com/apache/airflow/pull/25788))

Reviewer: ```suggestion
- Fixed an issue where the scheduler could crash when queueing dynamically mapped tasks [#25788](https://github.com/apache/airflow/pull/25788).
```
![image](https://user-images.githubusercontent.com/104205257/199715995-a5f9df73-d829-4d82-9106-063fff40fd66.png)
